src/StatsBase.jl

```diff
@@ -1,18 +1,34 @@
 __precompile__()
 
 module StatsBase
-    import Base: length, isempty, eltype, values, sum, mean, mean!, show, quantile
+    import Base: length, isempty, eltype, values, sum, show
     import Base.Cartesian: @nloops, @nref, @nextract
     using Base: @irrational, @propagate_inbounds
     import DataStructures: heapify!, heappop!, percolate_down!
+    using SortingAlgorithms
+    using Missings
+
+    if VERSION >= v"0.7.0-beta.85"
+        using Statistics
+        import Statistics: mean, mean!, var, varm, varm!, std, stdm, cov, covm,
+                           cor, corm, cov2cor!, unscaled_covzm, quantile, sqrt!,
+                           median, middle
+    else
+        import Base: mean, mean!, var, varm, varm!, std, stdm, cov, covm,
+                     cor, corm, cov2cor!, unscaled_covzm, quantile, sqrt!,
+                     median, middle
+    end
+
+    # Location of the `mean` function, used for calling the function inside of
+    # functions definitions that have `mean` as a keyword argument
+    const MEANHOME = (VERSION >= v"0.7.0-beta.85" ? Statistics : Base)::Module
```

Comment:
I don't really understand why it's still needed. It's used below when calling `mean` when it doesn't appear to be necessary.

Reply:
There are cases where we're calling the `mean` function when the keyword argument `mean` is `nothing`. In these cases we need to qualify the call to `mean` so that it doesn't pick up the keyword argument instead for the call. But what we qualify it with, Base or Statistics, depends on the version. Thus I stuck in this quick and dirty workaround to avoid version checks in every case of this.